src/main/scala/strawman/collection/immutable/Set.scala

```diff
@@ -22,4 +23,15 @@ trait SetMonoTransforms[A, +Repr]
 
   def - (elem: A): Repr
 
+}
+
+trait SetPolyTransforms[A, +C[X] <: Set[X] with SetLike[X, C]]
+  extends collection.SetPolyTransforms[A, C] {
+
+  protected def coll: C[A]
+
+  def ++ (that: collection.Set[A]): C[A] =
+    if (that.isEmpty) coll
+    else that.foldLeft[C[A]](coll)(_ + _)
```

Comment:
Actually, `++` uses `+`, which is notably slower in the strawman than in 2.12:

![cons](https://cloud.githubusercontent.com/assets/332812/24448810/8c419ba6-1475-11e7-9a97-ba456fefd82d.png)

I will investigate what’s wrong…